src/Elastic.Apm/DistributedTracing/TraceParent.cs

```diff
@@ -112,7 +106,10 @@ internal static bool TryExtractTraceparent(string traceParentValue, out string t
 
 			try
 			{
-				traceOptions = StringToByteArray(traceParentValue, VersionAndTraceIdAndSpanIdLength, OptionsLength);
+				var fields = StringToByteArray(traceParentValue, VersionAndTraceIdAndSpanIdLength, OptionsLength);
```

Comment:
You don't really need to allocate `fields` array, right? You can just have `StringToByte` which reads 2 chars from `start` index in `src` string.

Reply:
I did this, but the perf. difference is almost nothing - I measured it. This got it to `383.2 ns`... and reference less. I leave it like that, since it's still a plus.

Reply:
Could you please clarify what do you mean by?

> since it's still a plus

If you would like to keep `(Hex)StringToByteArray` for future use you can have it call `(Hex)StringToByte` to parse each byte.

It's not a big issue - I am just curious what pros you see in leaving the array. 
I personally see any cons - it makes the code less readable since the code does something that the spec doesn't require and it has performance impact.

Reply:
By "it's a plus" I meant that it's positive for performance, so it's better to keep it that way. So I guess we are on the same page :)

I agree, there is no reason to keep the array, and the current version already got rid of that, so I guess we are ok here. In case we need it later we can add it, but atm we don't need it, so I also think it's better to avoid that unnecessary allocation.